cmd/ci-reporter/cmd/github.go

```diff
@@ -0,0 +1,182 @@
+func GetGithubReportData(cfg ReporterConfig) (GithubReportData, error) {
+	ciSignalProjectBoard := []GithubProjectBoardColumn{newColumn, underInvestigationColumn}
+
+	// if the short flag is not set observingColumn & resolvedColumn will be added to the report
+	if !cfg.ShortReport {
+		ciSignalProjectBoard = append(ciSignalProjectBoard, observingColumn, resolvedColumn)
+	}
+
+	githubReportData := map[ColumnTitle][]IssueOverview{}
+	for _, column := range ciSignalProjectBoard {
+		cards, err := getCardsFromColumn(cfg, column.ColumnID)
+		if err != nil {
+			return nil, err
+		}
+		githubReportData[column.ColumnTitle] = cards
+	}
+	return githubReportData, nil
+}
+
+// PrintGithubReportData used to print github report data out to the console
+func PrintGithubReportData(reportData map[ColumnTitle][]IssueOverview) {
+	for columnTitle, issuesInColumn := range reportData {
+		if len(issuesInColumn) > 0 {
+			fmt.Println("\n" + columnTitle)
+			for _, issue := range issuesInColumn {
+				fmt.Printf("#%d %v\n - %s\n - %s\n", issue.ID, issue.Sigs, issue.Title, issue.URL)
+			}
+		}
+	}
+	fmt.Print("\n\n")
+}
+
+func getCardsFromColumn(cfg ReporterConfig, cardsID ColumnID) ([]IssueOverview, error) {
+	opt := &github.ProjectCardListOptions{}
+	cards, _, err := cfg.GithubClient.Projects.ListProjectCards(context.Background(), int64(cardsID), opt)
+	if err != nil {
+		fmt.Printf("error when querying cards %v", err)
+		return nil, err
+	}
+
+	issues := []IssueOverview{}
+	for _, c := range cards {
+		issueDetail, err := getIssueDetail(cfg, *c.ContentURL)
+		if err != nil {
+			return nil, err
+		}
+
+		overview := IssueOverview{
+			URL:   issueDetail.HTMLURL,
+			ID:    issueDetail.Number,
+			Title: issueDetail.Title,
+		}
+		for _, v := range issueDetail.Labels {
+			sig := sigRegex.FindString(*v.Name)
+			if sig != "" {
+				sig = strings.Replace(sig, "/", " ", 1)
+				overview.Sigs = append(overview.Sigs, sig)
+			}
+		}
+		issues = append(issues, overview)
+	}
+	return issues, nil
+}
+
+type issueDetail struct {
+	Number  int64          `json:"number"`
+	HTMLURL string         `json:"html_url"`
+	Title   string         `json:"title"`
+	Labels  []github.Label `json:"labels,omitempty"`
+}
+
+func getIssueDetail(cfg ReporterConfig, url string) (*issueDetail, error) {
+	// Create a new request using http
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	// add authorization header to the req
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", cfg.GithubToken))
+
+	// Send req using http Client
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Println("Error on response.\n[ERROR] -", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Printf("%v", err)
+		return nil, err
+	}
+	var result issueDetail
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
```

Comment:
Since the struct is relatively simpler and we don't modify it in the `getCardsFromColumn` function, should we return the value instead of a pointer?

Reply:
Thank you for that, this relates to the version that alena provided since this is part of the github reporter I will come back to that in one of the next PRs and check out which approach we can take.